server/lib/github-client.js

```diff
@@ -1,12 +1,23 @@
-const host = 'http://cyf-github-api.herokuapp.com';
 const fetch = require('isomorphic-fetch');
+const gitApi = 'https://api.github.com';
 
-function fetchFromGithub (path) {
-	// write something that will fetch json information from github
+fetchFromGithub = path => {
+  return (
+    fetch(`${gitApi}${path}`)
+    .then(apiRes => apiRes.json())
+    .then(json => json)
+    .catch(err => console.error(err))
+    );
 }
 
-module.exports.getUserProfile = function (username) {
-	return fetchFromGithub(`/users/${username}`);
-};
+filterByEventType = (data, eventType) => data.filter(value => value.type === eventType);
 
-// add a method to retrieve user pull requests here
+module.exports = {
+  getUserProfile: username => fetchFromGithub(`/users/${username}`),
+  getUserPullRequests: username => {
+  return (
+    fetchFromGithub(`/users/${username}/events/public`)
+    .then(data => filterByEventType(data, 'PullRequestEvent'))
```

Comment:
👍 this is much nicer. We _could_ even go further by implementing something called currying. This is when you have a function that returns another function which requires more data. This would be an ideal use case for currying. I'll try and speak to you today about it